processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/BlacklistedKeysFilter.java

```diff
@@ -37,11 +38,18 @@ public BlacklistedKeysFilter(ProcessorProperties props) {
              .listen((oldValue, newValue) -> ignoreKeys = new HashSet<>(newValue));
     }
 
-    public boolean shouldTake(ConsumerRecord<String, byte[]> record) {
+    public boolean shouldTake(ConsumerRecord<byte[], byte[]> record) {
+        final byte[] key = record.key();
+        if (key == null) {
+            return true;
+        }
+
+        final String stringKey = new String(key, StandardCharsets.UTF_8);
```

Comment:
what if the key cannot be read a String? this can happen. https://stackoverflow.com/questions/70667113/why-cant-i-decode-any-byte-using-utf-8#comment124924879_70667113

it looks like `ignoreKeys` should keep a set of byte arrays instead

Reply:
I see. It seems we might want to support other input for this denylist, maybe longs...
Let's leave the possibility open even if we don't implement it in this PR